### PR TITLE
`rotary_transform` -> `cartesian_to_rotary`, `rotary_to_cartesian`

### PR DESCRIPTION
**Pull request by selipot**
This PR renames `signal.rotary_transform` to `cartesian_to_rotary` and introduces `rotary_to_cartesian` as its invert. 

Are we happy with the input/output types? Should we test/allows pandas series as well?

### CODE REVIEW

**Review by milancurcic — APPROVED** [review]
Pending Jonathan's suggestions for change (if any), as I don't fully understand these algorithms, these changes all look reasonable to me.

**Comment by selipot** [comment]
I redid the implementation following latest conversation with @jonathanlilly.

File: clouddrift/signal.py [hunk 1]
```diff
@@ -7,56 +7,85 @@
...
+    References
+    ----------
+    [1] Gabor D. 1946 Theory of communication. Proc. IEE 93, 429–457. (10.1049/ji-1.1947.0015).
+
+    [2] Lilly JM, Olhede SC. 2010 Bivariate instantaneous frequency and bandwidth.
+    IEEE T. Signal Proces. 58, 591–603. (10.1109/TSP.2009.2031729).
+
+    See Also
+    --------
+    :func:`rotary_to_cartesian`, `cartesian_to_rotary`
```
Comment (milancurcic) [inline clouddrift/signal.py:88]:
```suggestion
    :func:`rotary_to_cartesian`, :func:`cartesian_to_rotary`
```

File: clouddrift/signal.py [hunk 2]
```diff
@@ -89,119 +119,187 @@ def analytic_transform(
...
 ) -> Tuple[np.ndarray, np.ndarray]:
-    """Return time-domain rotary components time series (zp,zn) from Cartesian components time series (u,v).
-    Note that zp and zn are both analytic time series which implies that the complex-valued time series
-    u+1j*v is recovered by zp+np.conj(zn). The mean of the original complex signal is split evenly between
-    the two rotary components.
+    """Return rotary signals (wp,wn) from analytic Cartesian signals (ua,va).
+
+    If ua is the analytic signal from real-valued signal u, and va the analytic signal
+    from real-valued signal v, then the positive (counterclockwise) and negative (clockwise)
+    signals are defined by wp = 0.5*(up+1j*vp), wp = 0.5*(up-1j*vp).
```
Comment (milancurcic) [inline clouddrift/signal.py:170]: We currently don't have a convention whether to put code snippets in docstrings in code blocks (double backticks) quotes or not. Using code blocks makes them look nicer in HTML docs, but less nice in interpreter. We don't have to resolve this now but let's think about it and be consistent.

**Review by milancurcic — APPROVED** [review]
Looks good.

**Review by philippemiron — APPROVED** [review]

**Pull request merged**